tests/general/process_test.go

```diff
@@ -0,0 +1,100 @@
+package tests
+
+import (
+	"io/ioutil"
+	"log"
+	"net/http"
+	"path"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zaptest/observer"
+
+	"github.com/signalfx/splunk-otel-collector/tests/testutils"
+)
+
+func TestCollectorProcessWithMultipleConfigs(t *testing.T) {
+	logCore, logs := observer.New(zap.DebugLevel)
+	logger := zap.New(logCore)
+	collector, err := testutils.NewCollectorProcess().
+		WithArgs("--config", path.Join(".", "testdata", "receivers.yaml"),
+			"--config", path.Join(".", "testdata", "processors.yaml"),
+			"--config", path.Join(".", "testdata", "exporters.yaml"),
+			"--config", path.Join(".", "testdata", "services.yaml")).
+		WithLogger(logger).
+		Build()
+
+	require.NotNil(t, collector)
+	require.NoError(t, err)
+
+	err = collector.Start()
+	require.NoError(t, err)
+
+	_, ok := collector.(*testutils.CollectorProcess)
+	require.True(t, ok)
+
+	require.Eventually(t, func() bool {
+		for _, log := range logs.All() {
+			if strings.Contains(log.Message,
+				`Set config to [testdata/receivers.yaml,testdata/processors.yaml,testdata/exporters.yaml,testdata/services.yaml]`,
+			) {
+				return true
+			}
+		}
+		return false
+	}, 20*time.Second, time.Second)
+
+	require.Eventually(t, func() bool {
+		for _, log := range logs.All() {
+			// Confirm collector starts and runs successfully
+			if strings.Contains(log.Message, "Everything is ready. Begin running and processing data.") {
+				return true
+			}
+		}
+		return false
+	}, 20*time.Second, time.Second)
+
+	require.Eventually(t, func() bool {
+		// Need expected configs for comparison to config server contents. Sections in the config server can be in
```

Comment:
This seems ok for now since an unrelated issue but ideally we'd have the config server behave more reliably.

Reply:
More than anything else this eventually loop is to just brush over the timing of starting the collector and the config server. The config server is expected to be up and running by the time we see the "Everything is ready" message, but since these are asynchronous methods, we'd have to wait for the first to finish successfully and then run this one a single time. I'm sure it could be done but I didn't think it was important enough to spend more time trying to figure it out for a test.